### PR TITLE
Feat/3 auth

### LINKED ISSUE #3: [Feat] auth 생성
## ✨작성할 기능에 대한 설명을 적어주세요
```
Spring Security Configuration 설정
JWT토큰발급을 위한 Member 생성 api , Member 조회 API생성
JwtProvider setting
Redis를 이용하여 JWT토큰 생성 및 재발급
Swagger로 JWT토큰 테스트
```

<br><br>

## ✅ 해야 할 작업들을 작성해주세요
- [x]  Spring Security Configuration 설정
- [x]  JWT토큰발급을 위한 Member 생성 api , Member 조회 API생성
- [x]  Redis를 이용하여 JWT토큰 생성 및 재발급
- [x] JwtProvider setting

<br><br>

## 🦋 연관된 이슈
-

### PR DESCRIPTION
**Pull request by minwoo1999**
## ✨ 요약
```
Spring Security Configuration 설정
JWT토큰발급을 위한 Member 생성 api , Member 조회 API생성
JwtProvider setting
Redis를 이용하여 JWT토큰 생성 및 재발급
Swagger로 JWT토큰 테스트

TMI::
원규형님이 TODO쪽에서 MemberId를 많이 사용하실 것 같아서 SecurityUtil클래스 따로 빼서
getCurrentUserId로 현재 로그인 중인 memberId 값 바로 가져올 수 있도록 Util함수 생성해봤습니다
```

<br><br>

## 😎 해결한 이슈
- close #3 

<br><br>

### CODE REVIEW

**Comment by minwoo1999** [comment]
### 테스트 커버리지 측정
|File|Coverage [51.72%]|:green_apple:|
|:-|:-:|:-:|
|[MemberDetails.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FMemberDetails.java)|100%|:green_apple:|
|[CookieUtil.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fcookie%2FCookieUtil.java)|100%|:green_apple:|
|[ErrorCodes.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fglobal%2FErrorCodes.java)|100%|:green_apple:|
|[MemberRole.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2FMemberRole.java)|91.43%|:green_apple:|
|[ErrorResponse.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fglobal%2FErrorResponse.java)|86.89%|:green_apple:|
|[RefreshTokenService.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fservice%2FRefreshTokenService.java)|86.84%|:green_apple:|
|[Member.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fdomain%2FMember.java)|82.57%|:green_apple:|
|[MemberDetailsService.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FMemberDetailsService.java)|80.65%|:green_apple:|
|[JwtAuthenticationFilter.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FJwtAuthenticationFilter.java)|80.15%|:green_apple:|
|[MemberResponseDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fapplication%2Fdto%2FMemberResponseDto.java)|72.73%|:green_apple:|
|[RefreshToken.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fentity%2FRefreshToken.java)|67.27%|:green_apple:|
|[CommonResponse.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fglobal%2Fdto%2FCommonResponse.java)|66.93%|:green_apple:|
|[MemberServiceImpl.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fapplication%2Fimpl%2FMemberServiceImpl.java)|63.12%|:green_apple:|
|[MemberController.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fpresentation%2FMemberController.java)|61.34%|:green_apple:|
|[MemeberRequestDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fapplication%2Fdto%2FMemeberRequestDto.java)|51.63%|:green_apple:|
|[JwtAuthorizationFilter.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FJwtAuthorizationFilter.java)|51.43%|:green_apple:|
|[JwtProvider.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FJwtProvider.java)|41.83%|:green_apple:|
|[GlobalExceptionHandlerFilter.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fglobal%2FGlobalExceptionHandlerFilter.java)|40.32%|:green_apple:|
|[ReworkApplication.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2FReworkApplication.java)|37.5%|:green_apple:|
|[CommonResDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fglobal%2Fcommon%2FCommonResDto.java)|27.63%|:x:|
|[BaseTimeEntity.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fglobal%2Fbase%2FBaseTimeEntity.java)|25%|:x:|
|[DuplicateAccountException.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fglobal%2Ferror%2FDuplicateAccountException.java)|25%|:x:|
|[GlobalExceptionHandler.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fglobal%2FGlobalExceptionHandler.java)|13.22%|:x:|
|[CommonResponse.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fexception%2Fdto%2FCommonResponse.java)|0%|:x:|
|[ErrorResponse.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fglobal%2Fdto%2FErrorResponse.java)|0%|:x:|
|[PasswordNotMatchException.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fglobal%2Ferror%2FPasswordNotMatchException.java)|0%|:x:|
|[UnAuthorizedException.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fglobal%2Ferror%2FUnAuthorizedException.java)|0%|:x:|
|[EncryptException.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fglobal%2Ferror%2FEncryptException.java)|0%|:x:|
|[NotFoundAccountException.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fglobal%2Ferror%2FNotFoundAccountException.java)|0%|:x:|
|[InvalidTokenException.java](https://github.com/rework-kr/REWORK-SERVER/blob/a9f5df7a8caa8ad8832e0999594ce0210f235db5/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fglobal%2Ferror%2FInvalidTokenException.java)|0%|:x:|

|Total Project Coverage|51.72%|:green_apple:|
|:-|:-:|:-:|

**Comment by minwoo1999** [comment]
### 테스트 커버리지 측정
|File|Coverage [60.47%]|:green_apple:|
|:-|:-:|:-:|
|[MemberDetails.java](https://github.com/rework-kr/REWORK-SERVER/blob/5f725932e2b2034c2336b63d596e0af20464bcc1/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FMemberDetails.java)|100%|:green_apple:|
|[CookieUtil.java](https://github.com/rework-kr/REWORK-SERVER/blob/5f725932e2b2034c2336b63d596e0af20464bcc1/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fcookie%2FCookieUtil.java)|100%|:green_apple:|
|[MemberRole.java](https://github.com/rework-kr/REWORK-SERVER/blob/5f725932e2b2034c2336b63d596e0af20464bcc1/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2FMemberRole.java)|91.43%|:green_apple:|
|[RefreshTokenService.java](https://github.com/rework-kr/REWORK-SERVER/blob/5f725932e2b2034c2336b63d596e0af20464bcc1/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fservice%2FRefreshTokenService.java)|86.84%|:green_apple:|
|[Member.java](https://github.com/rework-kr/REWORK-SERVER/blob/5f725932e2b2034c2336b63d596e0af20464bcc1/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fdomain%2FMember.java)|82.57%|:green_apple:|
|[MemberDetailsService.java](https://github.com/rework-kr/REWORK-SERVER/blob/5f725932e2b2034c2336b63d596e0af20464bcc1/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FMemberDetailsService.java)|80.65%|:green_apple:|
|[JwtAuthenticationFilter.java](https://github.com/rework-kr/REWORK-SERVER/blob/5f725932e2b2034c2336b63d596e0af20464bcc1/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FJwtAuthenticationFilter.java)|80.15%|:green_apple:|
|[MemberResponseDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/5f725932e2b2034c2336b63d596e0af20464bcc1/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fapplication%2Fdto%2FMemberResponseDto.java)|72.73%|:green_apple:|
|[RefreshToken.java](https://github.com/rework-kr/REWORK-SERVER/blob/5f725932e2b2034c2336b63d596e0af20464bcc1/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fentity%2FRefreshToken.java)|67.27%|:green_apple:|
|[MemberServiceImpl.java](https://github.com/rework-kr/REWORK-SERVER/blob/5f725932e2b2034c2336b63d596e0af20464bcc1/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fapplication%2Fimpl%2FMemberServiceImpl.java)|63.12%|:green_apple:|
|[MemberController.java](https://github.com/rework-kr/REWORK-SERVER/blob/5f725932e2b2034c2336b63d596e0af20464bcc1/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fpresentation%2FMemberController.java)|61.34%|:green_apple:|
|[MemeberRequestDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/5f725932e2b2034c2336b63d596e0af20464bcc1/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fapplication%2Fdto%2FMemeberRequestDto.java)|51.63%|:green_apple:|
|[JwtAuthorizationFilter.java](https://github.com/rework-kr/REWORK-SERVER/blob/5f725932e2b2034c2336b63d596e0af20464bcc1/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FJwtAuthorizationFilter.java)|51.43%|:green_apple:|
|[JwtProvider.java](https://github.com/rework-kr/REWORK-SERVER/blob/5f725932e2b2034c2336b63d596e0af20464bcc1/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FJwtProvider.java)|41.83%|:green_apple:|
|[ReworkApplication.java](https://github.com/rework-kr/REWORK-SERVER/blob/5f725932e2b2034c2336b63d596e0af20464bcc1/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2FReworkApplication.java)|37.5%|:green_apple:|
|[CommonResponse.java](https://github.com/rework-kr/REWORK-SERVER/blob/5f725932e2b2034c2336b63d596e0af20464bcc1/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fexception%2Fdto%2FCommonResponse.java)|0%|:x:|

|Total Project Coverage|60.47%|:green_apple:|
|:-|:-:|:-:|

**Comment by minwoo1999** [comment]
### 테스트 커버리지 측정
|File|Coverage [60.47%]|:green_apple:|
|:-|:-:|:-:|
|[MemberDetails.java](https://github.com/rework-kr/REWORK-SERVER/blob/db7e5b6a42493b0dd1d4c4f887b32f45c8d9a173/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FMemberDetails.java)|100%|:green_apple:|
|[CookieUtil.java](https://github.com/rework-kr/REWORK-SERVER/blob/db7e5b6a42493b0dd1d4c4f887b32f45c8d9a173/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fcookie%2FCookieUtil.java)|100%|:green_apple:|
|[MemberRole.java](https://github.com/rework-kr/REWORK-SERVER/blob/db7e5b6a42493b0dd1d4c4f887b32f45c8d9a173/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2FMemberRole.java)|91.43%|:green_apple:|
|[RefreshTokenService.java](https://github.com/rework-kr/REWORK-SERVER/blob/db7e5b6a42493b0dd1d4c4f887b32f45c8d9a173/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fservice%2FRefreshTokenService.java)|86.84%|:green_apple:|
|[Member.java](https://github.com/rework-kr/REWORK-SERVER/blob/db7e5b6a42493b0dd1d4c4f887b32f45c8d9a173/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fdomain%2FMember.java)|82.57%|:green_apple:|
|[MemberDetailsService.java](https://github.com/rework-kr/REWORK-SERVER/blob/db7e5b6a42493b0dd1d4c4f887b32f45c8d9a173/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FMemberDetailsService.java)|80.65%|:green_apple:|
|[JwtAuthenticationFilter.java](https://github.com/rework-kr/REWORK-SERVER/blob/db7e5b6a42493b0dd1d4c4f887b32f45c8d9a173/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FJwtAuthenticationFilter.java)|80.15%|:green_apple:|
|[MemberResponseDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/db7e5b6a42493b0dd1d4c4f887b32f45c8d9a173/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fapplication%2Fdto%2FMemberResponseDto.java)|72.73%|:green_apple:|
|[RefreshToken.java](https://github.com/rework-kr/REWORK-SERVER/blob/db7e5b6a42493b0dd1d4c4f887b32f45c8d9a173/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fentity%2FRefreshToken.java)|67.27%|:green_apple:|
|[MemberServiceImpl.java](https://github.com/rework-kr/REWORK-SERVER/blob/db7e5b6a42493b0dd1d4c4f887b32f45c8d9a173/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fapplication%2Fimpl%2FMemberServiceImpl.java)|63.12%|:green_apple:|
|[MemberController.java](https://github.com/rework-kr/REWORK-SERVER/blob/db7e5b6a42493b0dd1d4c4f887b32f45c8d9a173/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fpresentation%2FMemberController.java)|61.34%|:green_apple:|
|[MemeberRequestDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/db7e5b6a42493b0dd1d4c4f887b32f45c8d9a173/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fapplication%2Fdto%2FMemeberRequestDto.java)|51.63%|:green_apple:|
|[JwtAuthorizationFilter.java](https://github.com/rework-kr/REWORK-SERVER/blob/db7e5b6a42493b0dd1d4c4f887b32f45c8d9a173/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FJwtAuthorizationFilter.java)|51.43%|:green_apple:|
|[JwtProvider.java](https://github.com/rework-kr/REWORK-SERVER/blob/db7e5b6a42493b0dd1d4c4f887b32f45c8d9a173/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FJwtProvider.java)|41.83%|:green_apple:|
|[ReworkApplication.java](https://github.com/rework-kr/REWORK-SERVER/blob/db7e5b6a42493b0dd1d4c4f887b32f45c8d9a173/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2FReworkApplication.java)|37.5%|:green_apple:|
|[CommonResponse.java](https://github.com/rework-kr/REWORK-SERVER/blob/db7e5b6a42493b0dd1d4c4f887b32f45c8d9a173/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fexception%2Fdto%2FCommonResponse.java)|0%|:x:|

|Total Project Coverage|60.47%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/com/example/rework/config/security/SecurityUtils.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        }
+
+        if (authentication.isAuthenticated()
+                && !CollectionUtils.containsAny(
+                authentication.getAuthorities(), notUserAuthority)) {
+            return Long.valueOf(authentication.getName());
+        }
+        // admin 유저일시 유저 아이디 0 반환
+        return 0L;
+    }
```
Comment (OneK-2) [inline src/main/java/com/example/rework/config/security/SecurityUtils.java:30]: 굉장히 배려심 넘치는 부분 너무 감동적..😭

File: src/main/java/com/example/rework/global/ErrorCodes.java
Comment (OneK-2) [inline src/main/java/com/example/rework/global/ErrorCodes.java:1]: 이 부분 잘 참고해서 에러처리 해보겠습니다!

**Review by OneK-2 — APPROVED** [review]
너무 고생했어요~ :)
파일이 너무 많아서 이슈를 한번 나눴어야 했나 싶었는데, 다들 내용이 연결되어 있어서 나누기 애매했던 것 같네요. 결과적으로는 굳굳!
중간에 util만들어준거 완전 감동쓰..
### 궁금한점
1. 에러는 전부 GlobalExceptionHandler에 추가해주면 되는건가요?
2. 에러코드 쓰는 규칙 있으면 알려주세용
3. global에 있는 CommonResponse랑 member 밑에 있는 CommonResponse의 차이점이 뭔지 궁금합니다. 용도가 다른건가?
4. 내 로컬에서 서버 돌릴 때 추가로 설정해 줄 부분 뭐가 있는지도 궁금해요. 레디스 같은 부분 따로 설정해야 하는지.
5. 테스트 작성할때 jacoco 사용 관련 링크나 참고자료 추천 부탁해요!

File: src/main/java/com/example/rework/config/security/SecurityUtils.java (hunk 1 above)
Comment (minwoo1999) [reply src/main/java/com/example/rework/config/security/SecurityUtils.java:30]: ㅎㅎ 감사합니다

File: src/main/java/com/example/rework/global/ErrorCodes.java
Comment (minwoo1999) [reply src/main/java/com/example/rework/global/ErrorCodes.java:1]: > 이 부분 잘 참고해서 에러처리 해보겠습니다!

옙 :)

**Comment by minwoo1999** [comment]
궁금한점에 대한 답변
에러는 전부 GlobalExceptionHandler에 추가해주면 되는건가요?
-> 네네 맞습니다! 그부분에 직접 커스텀해서 만듬 exception들 추가해주시면 됩니다!

에러코드 쓰는 규칙 있으면 알려주세용
-> 저는 권한이 없으면 401, 허용되지않은 곳 403, 찾지 못했을 경우 404, 충돌 및 중복 409 , 나머지 서브 서버에러는 400 badrequest나
500때에러로 알 수 없는에러로 처리합니다! 이부분은 원규형님하고 나중에 한번 맞춰봐도 좋을 것 같네요!

global에 있는 CommonResponse랑 member 밑에 있는 CommonResponse의 차이점이 뭔지 궁금합니다. 용도가 다른건가?
- 헉 똑같네요, 저부분은 수정해서 member안에 있는 CommonResponse 클래스는 삭제하도록 하겠습니다 감사합니다 __ :)
내 로컬에서 서버 돌릴 때 추가로 설정해 줄 부분 뭐가 있는지도 궁금해요. 레디스 같은 부분 따로 설정해야 하는지.
- 로컬에 redis만 셋팅해주시면 될 것같아요! 
- docker desktop 설치 -> docker pull redis -> docker run redis 하신다음에 스프링부트 키시면 바로 적용될 것 같습니다~
테스트 작성할때 jacoco 사용 관련 링크나 참고자료 추천 부탁해요!
- 카톡으로 보냄 완료!

**Comment by minwoo1999** [comment]
### 테스트 커버리지 측정
|File|Coverage [65.83%]|:green_apple:|
|:-|:-:|:-:|
|[MemberDetails.java](https://github.com/rework-kr/REWORK-SERVER/blob/8989274d3c984f55fc4d3a6aa0782e09b8fd583f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FMemberDetails.java)|100%|:green_apple:|
|[CookieUtil.java](https://github.com/rework-kr/REWORK-SERVER/blob/8989274d3c984f55fc4d3a6aa0782e09b8fd583f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fcookie%2FCookieUtil.java)|100%|:green_apple:|
|[MemberRole.java](https://github.com/rework-kr/REWORK-SERVER/blob/8989274d3c984f55fc4d3a6aa0782e09b8fd583f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2FMemberRole.java)|91.43%|:green_apple:|
|[RefreshTokenService.java](https://github.com/rework-kr/REWORK-SERVER/blob/8989274d3c984f55fc4d3a6aa0782e09b8fd583f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fservice%2FRefreshTokenService.java)|86.84%|:green_apple:|
|[Member.java](https://github.com/rework-kr/REWORK-SERVER/blob/8989274d3c984f55fc4d3a6aa0782e09b8fd583f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fdomain%2FMember.java)|82.57%|:green_apple:|
|[MemberDetailsService.java](https://github.com/rework-kr/REWORK-SERVER/blob/8989274d3c984f55fc4d3a6aa0782e09b8fd583f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FMemberDetailsService.java)|80.65%|:green_apple:|
|[JwtAuthenticationFilter.java](https://github.com/rework-kr/REWORK-SERVER/blob/8989274d3c984f55fc4d3a6aa0782e09b8fd583f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FJwtAuthenticationFilter.java)|80.15%|:green_apple:|
|[MemberResponseDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/8989274d3c984f55fc4d3a6aa0782e09b8fd583f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fapplication%2Fdto%2FMemberResponseDto.java)|72.73%|:green_apple:|
|[RefreshToken.java](https://github.com/rework-kr/REWORK-SERVER/blob/8989274d3c984f55fc4d3a6aa0782e09b8fd583f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fentity%2FRefreshToken.java)|67.27%|:green_apple:|
|[MemberServiceImpl.java](https://github.com/rework-kr/REWORK-SERVER/blob/8989274d3c984f55fc4d3a6aa0782e09b8fd583f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fapplication%2Fimpl%2FMemberServiceImpl.java)|63.12%|:green_apple:|
|[MemberController.java](https://github.com/rework-kr/REWORK-SERVER/blob/8989274d3c984f55fc4d3a6aa0782e09b8fd583f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fpresentation%2FMemberController.java)|61.34%|:green_apple:|
|[MemeberRequestDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/8989274d3c984f55fc4d3a6aa0782e09b8fd583f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmember%2Fapplication%2Fdto%2FMemeberRequestDto.java)|51.63%|:green_apple:|
|[JwtAuthorizationFilter.java](https://github.com/rework-kr/REWORK-SERVER/blob/8989274d3c984f55fc4d3a6aa0782e09b8fd583f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FJwtAuthorizationFilter.java)|51.43%|:green_apple:|
|[JwtProvider.java](https://github.com/rework-kr/REWORK-SERVER/blob/8989274d3c984f55fc4d3a6aa0782e09b8fd583f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fauth%2Fjwt%2FJwtProvider.java)|41.83%|:green_apple:|
|[ReworkApplication.java](https://github.com/rework-kr/REWORK-SERVER/blob/8989274d3c984f55fc4d3a6aa0782e09b8fd583f/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2FReworkApplication.java)|37.5%|:green_apple:|

|Total Project Coverage|65.83%|:green_apple:|
|:-|:-:|:-:|

**Review by OneK-2 — APPROVED** [review]
고생하셨습니다!

**Pull request merged**